src/cider/exec/processor/DefaultJoinHashTableBuilder.cpp

```diff
@@ -23,13 +23,23 @@
 
 namespace cider::exec::processor {
 
+// TODO: pass some arguments that will decide hashtable type
 std::unique_ptr<JoinHashTable> DefaultJoinHashTableBuilder::build() {
-  // TODO: build a join hash table
+  // TODO: get the choosed hashtable type
+  // reset to other hashtable if changed, comment out due to only have one hashtable now
+  // hashTable_.reset(new JoinHashTable());
   return std::move(hashTable_);
 }
-
-void DefaultJoinHashTableBuilder::appendBatch(std::shared_ptr<CiderBatch> batch) {
-  // TODO: add batch into hashTable
+// TODO: get the join key. Right use hard-code col 0
+void DefaultJoinHashTableBuilder::appendBatch(
+    std::shared_ptr<cider::exec::nextgen::context::Batch> batch) {
+  int length = batch->getArray()->children[0]->length;
+  for (int i = 0; i < length; i++) {
+    std::any key = *((reinterpret_cast<int*>(
+                         const_cast<void*>(batch->getArray()->children[0]->buffers[1]))) +
+                     i);
+    hashTable_->getHashTable()->emplace(key, std::make_pair(batch.get(), i));
```

Comment:
possible to make API like `hashTable_>emplace(key, std::make_pair(batch.get(), i));`?

Reply:
update